disnake/ext/commands/base_core.py

```diff
@@ -648,7 +648,19 @@ async def can_run(self, inter: ApplicationCommandInteraction) -> bool:
             inter.application_command = original
 
 
-def default_member_permissions(value: int = 0, **permissions: Literal[True]) -> Callable[[T], T]:
+@overload
+def default_member_permissions(value: bool) -> Callable[[T], T]:
+    ...
+
+
+@overload
+def default_member_permissions(value: int = 0, **permissions: bool) -> Callable[[T], T]:
+    ...
+
+
+def default_member_permissions(
+    value: Union[int, bool] = 0, **permissions: bool
+) -> Callable[[T], T]:
```

Comment:
Not sure about supporting value being a boolean. I don't think we should do this.

Reply:
535bc2372bce2c8df8bb152d960aff0f422c8d98